src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScriptTest.java

```diff
@@ -113,4 +116,33 @@ private String dumpError() {
         assertEquals(dumpError(), Result.SUCCESS, exec.getResult());
     }
 
+    private ErrorAction extractFirstErrorAction() {
+        StringBuilder msg = new StringBuilder();
+
+        FlowGraphWalker walker = new FlowGraphWalker(exec);
+        for (FlowNode n : walker) {
+            ErrorAction e = n.getAction(ErrorAction.class);
+            if (e != null) {
+                return e;
+            }
+        }
+        return null;
+    }
+
+    @Issue("JENKINS-34488")
+    @WithTimeout(30)
+    @Test public void unserializableException() throws Exception {
+        CpsFlowDefinition flow = new CpsFlowDefinition("assert false");
+        createExecution(flow);
+        exec.start();
+        while (!exec.isComplete()) {
+            exec.waitForSuspension();
+        }
+        assertEquals(Result.FAILURE, exec.getResult());
+
+        // extract ErrorAction.
+        ErrorAction e = extractFirstErrorAction();
+        assertNotNull(e);
+        assertThat(e.getError(), Matchers.instanceOf(ErrorActionException.class));
```

Comment:
Easier and more general to write this in pretty much any other test suite, using actual `WorkflowJob`s rather than `AbstractCpsFlowTest`, since you can just `assertLogContains` with whatever text you would want the build to print had this bug not existed (`Assertion failed`?).